compiler+runtime/src/cpp/jank/runtime/obj/atom.cpp

```diff
@@ -43,10 +45,25 @@ namespace jank::runtime::obj
     return val.load();
   }
 
+  static void notify_watches(atom_ref const a, object_ref const old_val, object_ref const new_val)
+  {
+    auto const locked_watches(a->watches.rlock());
+    for(auto const entry : (*locked_watches)->data)
```

Comment:
Just a question: does the order matter here? i.e. does Clojure promise any sort of ordering for notifications? If so, looping through the map will not provide stable ordering (as the map changes) and it will not persist the ordering in which the watches were added.

Reply:
Ok, sounds good. Thanks for looking it up! This is a good thing to leave in a comment, above the definition of `watches`, I think. Just as a friendly reminder to anyone in the code.